client/asset/eth/eth.go

```diff
@@ -2016,6 +2015,73 @@ func (w *TokenWallet) EstimateRegistrationTxFee(feeRate uint64) uint64 {
 		return math.MaxUint64
 	}
 	return g.Transfer * feeRate
+}
+
+// EstimateSendTxFee returns a tx fee estimate for sending provided amount using
+// the provided feeRate.
+func (w *ETHWallet) EstimateSendTxFee(value, _ uint64, subtract bool) (fee uint64, err error) {
+	if subtract {
+		return 0, fmt.Errorf("wallet does not support checking network fee for withdrawal")
+	}
+	bal, err := w.Balance()
+	if err != nil {
+		return 0, nil
+	}
+	avail := bal.Available
+	if avail < value {
+		return 0, fmt.Errorf("not enough funds to send: have %d gwei need %d gwei", avail, value)
+	}
+
+	maxFeeRate, err := w.recommendedMaxFeeRate(w.ctx)
+	if err != nil {
+		return 0, fmt.Errorf("error getting max fee rate: %w", err)
+	}
+
+	maxFee := defaultSendGasLimit * dexeth.WeiToGwei(maxFeeRate)
+	if avail < value+maxFee {
+		return 0, fmt.Errorf("not enough funds to send: cannot cover value %d, max fee of %d gwei", value, maxFee)
+	}
+
+	return maxFee, nil
+}
+
+// EstimateSendTxFee returns a tx fee estimate for sending provided amount using
+// the provided feeRate.
+func (w *TokenWallet) EstimateSendTxFee(value, _ uint64, subtract bool) (fee uint64, err error) {
```

Comment:
just curious on the reasons for calculating `feeRate` "from scratch" here (and in ETH wallet) instead of relying on wallet implementing `FeeRater` interface (and checking for 0 value it could return to report "fee estimation is currently unavailable, try later" error) ?

Reply:
It actually is already a FeeRater with `func (eth *baseWallet) FeeRate() uint64`

Fee estimation, as we have implemented it, is relatively simple for eth and should never be unavailable.

Reply:
I probably phrased my comment above quite vaguely,

wondering if `_ uint64` could actually be used in this method (and why it is ignored and fee is calculated "from scratch" instead)

Reply:
> wondering if _ uint64 could actually be used in this method (and why it is ignored and fee is calculated "from scratch" instead)

@buck54321, can you throw some light on this?

Reply:
`FeeRater` only returns a rate. To calculate the total fees, we have to know the tx size too, and to get the tx size, we need to consider what inputs are available to fund the tx.

Reply:
> wondering if `_ uint64` could actually be used in this method (and why it is ignored and fee is calculated "from scratch" instead)

This method just needs a comment: `// The provided fee rate is ignored since all sends will use an internally derived fee rate.`

Not all implementations of this interface method will behave that way, hence it is simply ignored by this impl.